src/wp_filter_sender.cpp

```diff
@@ -0,0 +1,244 @@
+	    ros::Duration(0.01).sleep();
+  	}
+  	if(current_state.mode == "GUIDED")
+	{
+		ROS_INFO("Mode set to GUIDED. Mission starting");
+		return 0;
+	}else{
+		ROS_INFO("Error starting mission!!");
+		return -1;	
+	}
+}
+
+int initialize_local_frame()
+{
+	//set the orientation of the local reference frame
+	ROS_INFO("Initializing local coordinate system");
+	local_offset = 0;
+	for (int i = 1; i <= 30; i++) {
+		ros::spinOnce();
+		ros::Duration(0.1).sleep();
+
+		
+
+		float q0 = current_pose.pose.pose.orientation.w;
+		float q1 = current_pose.pose.pose.orientation.x;
+		float q2 = current_pose.pose.pose.orientation.y;
+		float q3 = current_pose.pose.pose.orientation.z;
+		float psi = atan2((2*(q0*q3 + q1*q2)), (1 - 2*(pow(q2,2) + pow(q3,2))) ); // yaw
+
+		local_offset += psi*(180/M_PI);
+
+		local_offset_pose.x = local_offset_pose.x + current_pose.pose.pose.position.x;
+		local_offset_pose.y = local_offset_pose.y + current_pose.pose.pose.position.y;
+		local_offset_pose.z = local_offset_pose.z + current_pose.pose.pose.position.z;
+		// ROS_INFO("current heading%d: %f", i, local_offset_g/i);
+	}
+	local_offset_pose.x = local_offset_pose.x/30;
+	local_offset_pose.y = local_offset_pose.y/30;
+	local_offset_pose.z = local_offset_pose.z/30;
+	local_offset /= 30;
+	ROS_INFO("Coordinate offset set");
+	ROS_INFO("the X' axis is facing: %f", local_offset);
+	return 0;
+}
+
+int arm()
+{
+	//intitialize first waypoint of mission
+	set_destination(0,0,0,0,correction_heading, local_desired_heading, correction_vector, local_offset_pose, desired_waypoint);
+	for(int i=0; i<100; i++)
+	{
+		local_pos_pub.publish(desired_waypoint);
+		ros::spinOnce();
+		ros::Duration(0.01).sleep();
+	}
+	// arming
+	ROS_INFO("Arming drone");
+	mavros_msgs::CommandBool arm_request;
+	arm_request.request.value = true;
+	while (!current_state.armed && !arm_request.response.success && ros::ok())
+	{
+		ros::Duration(.1).sleep();
+		arming_client.call(arm_request);
+		local_pos_pub.publish(desired_waypoint);
+	}
+	if(arm_request.response.success)
+	{
+		ROS_INFO("Arming Successful");	
+		return 0;
+	}else{
+		ROS_INFO("Arming failed with %d", arm_request.response.success);
+		return -1;	
+	}
+}
+
+
+
+int init_publisher_subscriber(ros::NodeHandle controlnodehandle)
+{
+	local_pos_pub = controlnodehandle.advertise<geometry_msgs::PoseStamped>("mavros/setpoint_position/local", 10);
+	currentPos = controlnodehandle.subscribe<nav_msgs::Odometry>("/mavros/global_position/local", 10, pose_cb);
+	state_sub = controlnodehandle.subscribe<mavros_msgs::State>("mavros/state", 10, state_cb);
+	arming_client = controlnodehandle.serviceClient<mavros_msgs::CommandBool>("mavros/cmd/arming");
+	set_mode_client = controlnodehandle.serviceClient<mavros_msgs::SetMode>("/mavros/set_mode");
+    //goal_pos_sub =controlnodehandle.subscribe<task_master::TaskGoalPosition>("/hi",10,pose);
+    
+}
+
+
+int main(int argc, char** argv)
+{
+	//initialize ros 
+	ros::init(argc, argv, "wp_filter_sender");
```

Comment:
This code needs to be in every node so that we can change the logging level.     if (ros::console::set_logger_level(ROSCONSOLE_DEFAULT_NAME, ros::console::levels::Info))
        ros::console::notifyLoggerLevelsChanged();. Needs to go after ros::init... See https://github.com/IcebergASV/nav_channel/blob/main/src/nav_channel_ctrl.cpp for an example